src/main/java/org/peergos/PeriodicBlockProvider.java

```diff
@@ -15,55 +15,58 @@
 public class PeriodicBlockProvider {
 
     private static final Logger LOG = Logger.getLogger(PeriodicBlockProvider.class.getName());
-    private final long reprovideIntervalMillis;
-    private final Supplier<Stream<Cid>> getBlocks;
     private final Host us;
     private final Kademlia dht;
     private final Queue<Cid> newBlocksToPublish;
 
+    private final ControlSubThread cidProvider;
+
+    private final ControlSubThread newCidProvider;
```

Comment:
Is there a way to put a generic type on these to make it easier to tell what it is doing from the type, like before?